mediagrains/hypothesis/strategies_new.py

```diff
@@ -0,0 +1,615 @@
+# Copyright 2018 British Broadcasting Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""\
+Contains a number of useful hypothesis strategies which can be used to
+generate mediagrains for hypothesis based testing.
+
+This is for the 'new style' of instantiating grains.
+"""
+
+from mediatimestamp.hypothesis.strategies import immutabletimestamps as timestamps
+from hypothesis.strategies import (
+    integers,
+    from_regex,
+    booleans,
+    just,
+    fractions,
+    binary,
+    lists,
+    fixed_dictionaries,
+    one_of,
+    SearchStrategy,
+    builds,
+    sampled_from,
+    floats
+)
+
+import struct
+
+from uuid import UUID
+from fractions import Fraction
+from copy import copy
+
+from ..grains import new_attributes_for_grain_type as attributes_for_grain_type
+from ..cogenums import CogAudioFormat, CogFrameFormat, CogFrameLayout
+from ..grains import Grain, EventGrain, AudioGrain, CodedAudioGrain, CodedVideoGrain, VideoGrain
+
+
+__all__ = ["DONOTSET",
+           "empty_grains",
+           "event_grains",
+           "audio_grains",
+           "video_grains",
+           "coded_audio_grains",
+           "coded_video_grains",
+           "grains",
+           "grains_from_template_with_data",
+           "strategy_for_grain_attribute",
+           "shrinking_uuids",
+           "fraction_dicts",
+           "grains_with_data"]
+
+
+DONOTSET = object()
+
+
+def shrinking_uuids():
+    """A strategy that produces uuids, but shrinks towards 0, unlike the standard hypothesis one."""
+    return binary(min_size=16, max_size=16).map(lambda b: UUID(bytes=b))
+
+
+def fraction_dicts(*args, **kwargs):
+    """A strategy that produces dictionaries of the form {'numerator': n, 'denominator': d} for fractions generated
+    using the fractions strategy. All arguments are passed through to the underlying call to fractions."""
+    def _fraction_to_dict(f):
+        return {'numerator': f.numerator,
+                'denominator': f.denominator}
+    return builds(_fraction_to_dict, fractions(*args, **kwargs))
```

Comment:
Nit: Since these two are identical in both files, it might be worth importing into one from the other? More generally, I assume the old strategies are going away at some point, which is why they don't have a new/old flag?